src/ZipStream.php

```diff
@@ -399,6 +399,17 @@ protected function addFileHeader($name, $opt, $meth, $crc, $zlen, $len) {
 		// create dos timestamp
 		$opt['time'] = isset($opt['time']) && !empty($opt['time']) ? $opt['time'] : time();
 		$dts         = $this->dostime($opt['time']);
+                
+                // general purpose bit flag
+                $genb = 0x00;
+
+		if (mb_check_encoding($name, "UTF-8") && !mb_check_encoding($name, "ASCII"))
```

Comment:
It's worth noting that this line would add the mbstring extension as a dependency. Most installations have it, but I know some of our production servers do not. I'd add `"ext-mbstring": "*"` to the composer.json file